src/maidsafe/routing/connection_manager.cc

```diff
@@ -146,42 +149,42 @@ void ConnectionManager::AddNode(optional<NodeInfo> assumend_node_info, EndpointP
   auto socket = pair_i->second;
   weak_ptr<crux::socket> weak_socket = socket;
 
-  socket->async_connect
-      (convert::ToBoost(eps.external),
-       [=](boost::system::error_code error) {
-         auto socket = weak_socket.lock();
-
-         if (!socket) return;
+  socket->async_connect(convert::ToBoost(eps.external), [=](boost::system::error_code error) {
+    auto socket = weak_socket.lock();
 
-         if (error) {
-           being_connected_.erase(endpoint);
-           return;
-         }
+    if (!socket)
+      return;
 
-         NodeInfo our_node_info(our_id_, our_fob_);
+    if (error) {
+      being_connected_.erase(endpoint);
+      return;
+    }
 
-         AsyncExchange(*socket, Serialise(our_node_info),
-           [=](boost::system::error_code error, Bytes data) {
-             auto socket = weak_socket.lock();
+    AsyncExchange(*socket, Serialise(our_fob_.name(), our_fob_.Serialise()),
```

Comment:
Yes I agree we should not have serialise for non public fobs. Think its OK to use Pmid now as routing node creates these so users of the lib are not encumbered or restricted to our types so much, which I like. Nice one